misc/toml-parser/src/main/java/io/ballerina/toml/validator/schema/SchemaDeserializer.java

```diff
@@ -162,6 +175,51 @@ private String parseOptionalStringFromJson(JsonObject jsonObject, String key) {
         }
         throw new JsonSchemaException(key + " should always be a string");
     }
+
+    private Map<String, String> parseOptionalMapFromJson(JsonObject jsonObject, String key) {
+        Map<String, String> customMessages = new HashMap<>();
+        JsonObject customMessageJson = jsonObject.getAsJsonObject(key);
+        if (customMessageJson == null) {
+            return new HashMap<>();
+        }
+        JsonElement type = customMessageJson.get(TYPE);
+        if (!(type == null || type.isJsonNull())) {
+            String customTypeMessage = type.getAsString();
+            customMessages.put(TYPE, customTypeMessage);
+        }
+
+        JsonElement pattern = customMessageJson.get(PATTERN);
+        if (!(pattern == null || pattern.isJsonNull())) {
+            String customPatternMessage = pattern.getAsString();
+            customMessages.put(PATTERN, customPatternMessage);
+        }
+
+        JsonElement required = customMessageJson.get(REQUIRED);
+        if (!(required == null || required.isJsonNull())) {
+            String customRequiredMessage = required.getAsString();
+            customMessages.put(REQUIRED, customRequiredMessage);
+        }
+
+        JsonElement additionalProperties = customMessageJson.get(ADDITIONAL_PROPERTIES);
+        if (!(additionalProperties == null || additionalProperties.isJsonNull())) {
+            String customAdditionalPropMessage = additionalProperties.getAsString();
+            customMessages.put(ADDITIONAL_PROPERTIES, customAdditionalPropMessage);
+        }
+
+        JsonElement minimum = customMessageJson.get(MINIMUM);
+        if (!(minimum == null || minimum.isJsonNull())) {
+            String customMinMessage = minimum.getAsString();
+            customMessages.put(MINIMUM, customMinMessage);
+        }
+
+        JsonElement maximum = customMessageJson.get(MAXIMUM);
+        if (!(maximum == null || maximum.isJsonNull())) {
+            String customMaxMessage = maximum.getAsString();
+            customMessages.put(MAXIMUM, customMaxMessage);
+        }
```

Comment:
Can we replace this code with follwoing method to avoid duplicates?
```java
private void addFieldToCustomMessagesMap(Map<String, String> customMessages, JsonObject customMessageJson,
            String field) {
        JsonElement type = customMessageJson.get(field);
        if (!(type == null || type.isJsonNull())) {
            String customTypeMessage = type.getAsString();
            customMessages.put(field, customTypeMessage);
        }
    }
```
Then you can invoke
```java
        addFieldToCustomMessagesMap(customMessages, customMessageJson, TYPE);
        addFieldToCustomMessagesMap(customMessages, customMessageJson, PATTERN);
        addFieldToCustomMessagesMap(customMessages, customMessageJson, REQUIRED);
        addFieldToCustomMessagesMap(customMessages, customMessageJson, ADDITIONAL_PROPERTIES);
        addFieldToCustomMessagesMap(customMessages, customMessageJson, MINIMUM);
        addFieldToCustomMessagesMap(customMessages, customMessageJson, MAXIMUM);